rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/NSCreationServlet.java

```diff
@@ -149,10 +154,21 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
                                                                                        toArray(policyParams),
                                                                                        toArray(policyFileParams));
 
-            if (jsonResult.equals("true")) {
-                jsonResult = createNonEscapedSimpleJsonPair("result", "true");
-            }
+            JSONValue json = JSONParser.parseStrict(jsonResult);
+            JSONObject obj = json.isObject();
+            if (obj != null) {
+                if (obj.containsKey("valid")) {
+                    if (((JSONBoolean) obj.get("valid")).booleanValue()) {
+                        jsonResult = createNonEscapedSimpleJsonPair("result", "true");
+                    } else {
+                        String errorMessage = obj.get("errorMessage").toString();
+                        write(response,
+                              createJavascriptPayload(callbackName,
+                                                      createEscapedSimpleJsonPair("errorMessage", errorMessage)));
 
+                    }
+                }
+            }
```

Comment:
the following write call seems to overwrite the write statement executed inside the else block.